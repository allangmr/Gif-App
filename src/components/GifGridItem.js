import React from 'react'
import Proptypes from 'prop-types'

const GifGridItem = ( {title, id,url} ) => {
    return (
        <div className="card animate__animated animate__backInLeft" key={id}>
            <img src={url} alt={title} height="300" width="300" />
        </div>
    )
}

GifGridItem.propTypes = {
    title: Proptypes.string.isRequired,
    id: Proptypes.number.isRequired,
    url: Proptypes.string.isRequired
}
export default GifGridItem

