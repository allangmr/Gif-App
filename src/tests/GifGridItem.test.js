import { shallow } from "enzyme";
import '@testing-library/jest-dom'
import GifGridItem from "../components/GifGridItem";

describe('Test of GifGridItem', () => {
    const Grid = {title:'Prueba', id:1, url:'google.com'}
    const {title, id,url}  = Grid;
    const wrapper = shallow(<GifGridItem title={title} id={id} url={url} />);
    test('should be show the component', () => {
            expect(wrapper).toMatchSnapshot();
    });

    test('should be have one image with some title', () => {
        const img = wrapper.find('img');
        expect(img.props().alt).toBe(title);
        expect(img.props().src).toBe(url);
    });

    test('debe de tener animate__backInLeft', () => {
        const div = wrapper.find('div');
        expect(div.props().className).toContain('animate__backInLeft');
    });

});