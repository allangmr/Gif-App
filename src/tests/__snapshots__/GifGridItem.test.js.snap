// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of GifGridItem should be show the component 1`] = `
<div
  className="card animate__animated animate__backInLeft"
  key="1"
>
  <img
    alt="Prueba"
    height="300"
    src="google.com"
    width="300"
  />
</div>
`;
